public/app/features/plugins/sql/datasource/SqlDatasource.ts

```diff
@@ -115,6 +127,50 @@ export abstract class SqlDatasource extends DataSourceWithBackend<SQLQuery, SQLO
     };
   }
 
+  query(request: DataQueryRequest<SQLQuery>): Observable<DataQueryResponse> {
+    // This logic reenables the previous SQL behavior regarding what databases are available for the user to query.
+    if (isSqlDatasourceDatabaseSelectionFeatureFlagEnabled()) {
+      const databaseIssue = this.checkForDatabaseIssue(request);
+
+      if (!!databaseIssue) {
+        const error = new Error(databaseIssue);
+        return throwError(() => error);
+      }
+    }
+
+    return super.query(request);
+  }
+
+  private checkForDatabaseIssue(request: DataQueryRequest<SQLQuery>) {
+    // If the datasource is Postgres and there is no default database configured - either never configured or removed - return a database issue.
+    if (this.type === 'postgres' && !this.preconfiguredDatabase) {
+      return `You do not currently have a default database configured for this data source. Postgres requires a default
+             database with which to connect. Please configure one through the Data Sources Configuration page, or if you
+             are using a provisioning file, update that configuration file with a default database.`;
+    }
+
+    // No need to check for database change/update issues if the datasource is being used in Explore.
+    if (request.app !== CoreApp.Explore) {
+      /*
+        If a preconfigured datasource database has been added/updated - and the user has built ANY number of queries using a 
+        database OTHER than the preconfigured one, return a database issue - since those databases are no longer available.
+        The user will need to update their queries to use the preconfigured database.
+      */
+      if (!!this.preconfiguredDatabase) {
+        for (const target of request.targets) {
+          // Test for database configuration change only if query was made in `builder` mode.
+          if (target.editorMode === EditorMode.Builder && target.dataset !== this.preconfiguredDatabase) {
+            return `The configuration for this Panel's data source has been modified. The previous database used in this Panel's
```

Comment:
I personally wouldn't write Panel with a capital P, but I'm not sure if there are any guidelines regarding this. Might be something to look into.

Reply:
Agreed. Updated,